account_invoice_pricelist/view/view.xml

```diff
@@ -1,58 +1,48 @@
+        <field name="arch" type="xml">
+            <field name="state" position="before">
+                <button type="object" name="button_update_prices_from_pricelist" string="Update prices from pricelist" attrs="{'invisible': ['|', ('pricelist_id', '=', False), ('state', 'not in', ['draft'])]}" />
             </field>
-        </record>
-
-        <record id="view_account_invoice_supplier_form" model="ir.ui.view">
-            <field name="model">account.invoice</field>
-            <field name="inherit_id" ref="account.invoice_supplier_form"/>
-            <field name="arch" type="xml">
-                <field name="user_id" position="after">
-                    <field name="pricelist_id"/>
-                </field>
+            <field name="user_id" position="after">
+                <field name="pricelist_id"/>
             </field>
-        </record>
-
-        <record id="view_account_invoice_search" model="ir.ui.view">
-            <field name="model">account.invoice</field>
-            <field name="inherit_id" ref="account.view_account_invoice_filter"/>
-            <field name="arch" type="xml">
-                <xpath expr="//filter[@string='Partner']" position="after">
-                    <filter string="Pricelist" context="{'group_by':'pricelist_id'}"/>
-                </xpath>
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context">{'partner_id': partner_id, 'price_type': context.get('price_type') or False, 'type': type, 'pricelist_id': pricelist_id}</attribute>
             </field>
-        </record>
-
-    </data>
-</openerp>
-
-
+        </field>
+    </record>
+
+    <record id="view_account_invoice_supplier_form" model="ir.ui.view">
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_supplier_form"/>
+        <field name="arch" type="xml">
+            <field name="state" position="before">
+                <button type="object" name="button_update_prices_from_pricelist"
+                        string="Update prices from pricelist"
+                        attrs="{'invisible': ['|', ('pricelist_id', '=', False), ('state', 'not in', ['draft'])]}" />
+            </field>
+            <field name="user_id" position="after">
+                <field name="pricelist_id"/>
+            </field>
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context">{'partner_id': partner_id, 'price_type': context.get('price_type') or False, 'type': type, 'pricelist_id': pricelist_id}</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_account_invoice_search" model="ir.ui.view">
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.view_account_invoice_filter"/>
+        <field name="arch" type="xml">
+            <xpath expr="//filter[@name='group_by_partner_id']" position="after">
```

Comment:
If this view is shared for both invoice types, we should hide it for supplier ones. For sure there's a context variable that allows this.